src/main/java/org/myrobotlab/programab/handlers/oob/OobProcessor.java

```diff
@@ -0,0 +1,93 @@
+package org.myrobotlab.programab.handlers.oob;
+
+import java.util.List;
+
+import org.myrobotlab.codec.CodecUtils;
+import org.myrobotlab.framework.Message;
+import org.myrobotlab.programab.models.Mrl;
+import org.myrobotlab.programab.models.Oob;
+import org.myrobotlab.service.ProgramAB;
+
+public class OobProcessor {
+
+  private static OobProcessor instance;
+  private transient ProgramAB programab;
+  protected int maxBlockTime = 2000;
+
+  private OobProcessor() {
+  }
+
+  public static OobProcessor getInstance(ProgramAB programab) {
+    if (instance == null) {
+      instance = new OobProcessor();
+      instance.programab = programab;
+    }
+    return instance;
+  }
+
+  public Message toMsg(Mrl mrl) {
+    Object[] data = null;
+    if (mrl.params != null) {
+      data = new Object[mrl.params.size()];
+      for (int i = 0; i < data.length; ++i) {
+        data[i] = mrl.params.get(i).trim();
```

Comment:
Looks like this will end up only working for String parameters. Any way we can add an XML property to the param tag defining the type? Then we can take the string of the param and deserialize to the given type

Reply:
Heh, its a bigger topic.
You and @kwatters are pro strong type.  I'm not against it, but putting the burden on a network json interface to always supply a type isn't "friendly" imo.  Especially, if the type is the actual concrete name of some language type.

Reply:
How about an optional type parameter, and if it's not present we assume string? Technically we could use the method cache to attempt auto cast to other primitives, but then we get into the whole "which overload to call" mess, would like to avoid repeating that

Reply:
Yeah, a possibility.  
But these are 2 different topics.  For this pr I'm forcing the notation in the aiml json to be an array of messages, for simplicity and the fact many of the actions desired in behaviors require multiple messages.  As far as the parameters, its a very important topic - but in this small pr the context is not the network api, but the embedded oob tags.  I'm just thinking this is a +1 step in the right direction, because previously it was "only" strings. This implementation should handle all primitives as well.